src/audiovideocontroller/DefaultAudioVideoController.ts

```diff
@@ -1103,18 +1110,25 @@ export default class DefaultAudioVideoController
       this.meetingSessionContext.videoUplinkBandwidthPolicy &&
       !this.meetingSessionContext.enableSimulcast
     ) {
-      const oldMaxBandwidth = this.meetingSessionContext.videoUplinkBandwidthPolicy.maxBandwidthKbps();
-
-      this.meetingSessionContext.videoUplinkBandwidthPolicy.setHasBandwidthPriority(
-        hasBandwidthPriority
-      );
-      const newMaxBandwidth = this.meetingSessionContext.videoUplinkBandwidthPolicy.maxBandwidthKbps();
+      if (this.meetingSessionContext.videoUplinkBandwidthPolicy.chooseEncodingParameters().size > 0) {
+        this.meetingSessionContext.videoUplinkBandwidthPolicy.setHasBandwidthPriority(
+          hasBandwidthPriority
+        );
+        await this.enforceBandwidthLimitationForSender(this.meetingSessionContext.videoUplinkBandwidthPolicy.maxBandwidthKbps());
```

Comment:
You're computing the encoding parameters twice: once here and once in `enforceBandwidth`.